### ISSUE TITLE
TDE in 1.90.0 is breaking operations in PostgreSQL db instance

### ISSUE THREAD

**Issue by xinyifly** [aliyun/terraform-provider-alicloud#2646]
### Terraform Version

```
Terraform v0.12.28
+ provider.alicloud v1.90.0
```

### Affected Resource(s)

- db_instance

### Terraform Configuration Files
```hcl
resource "alicloud_db_instance" "this" {
  vswitch_id   = alicloud_vswitch.private.id
  security_ips = [alicloud_vpc.this.cidr_block]

  instance_name    = "xxxxxxx"
  engine           = "PostgreSQL"
  engine_version   = "12.0"
  instance_storage = 20
  instance_type    = "pg.n2.medium.1"

  # instance_charge_type = "Prepaid"
  # auto_renew           = true
}
```

### Panic Output

```
Error: [ERROR] terraform-provider-alicloud/alicloud/resource_alicloud_db_instance.go:714:
[ERROR] terraform-provider-alicloud/alicloud/service_alicloud_rds.go:798: Resource pgm-uf61zf6z4hg5ze67 DescribeDBInstanceTDE Failed!!! [SDK alibaba-cloud-sdk-go ERROR]:
SDK.ServerError
ErrorCode: InstanceEngineType.NotSupport
Recommend:
RequestId: C7BC3017-03C9-4BFA-B390-069FA039D474
Message: The instance engine and type does not support operations.
```

### Expected Behavior
No panic

### Actual Behavior
Panicked

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. use 1.89.0 `terraform apply`
2. use 1.90.0 `terraform refresh`

**Comment by socotra69** [comment]
Hi,

I have the same issue with RDS Mysql instances.
It seems to be linked with the commit "perf: rds ssl and tde", where !IsExpectedErrors(err, ...) on a nil error returns true so it crashes ...

**Comment by socotra69** [comment]
The newly published version 1.90.1 correct this error

**Comment by martinkubrak** [comment]
This also happens with `alicloud_kvstore_instance` with provider version 1.200.0.
